core/lib/dal/src/blocks_dal.rs

```diff
@@ -423,16 +424,30 @@ impl BlocksDal<'_, '_> {
                 .with_arg("eth_tx_id", &eth_tx_id)
                 .execute(self.storage)
                 .await?;
+
+                if result.rows_affected() == 0 {
+                    let err = Instrumented::new("set_eth_tx_id#commit")
```

Comment:
Nit: You can define this `Instrumented` instance before the query and share it both for the query (via `Instrumented::with()`) and the following check. Would require minor changes to the `db_connection` crate (namely, adding `Clone` for `Instumented<'_, ()>`).